fix(frontend): forward Host header as X-Forwarded-Host

codesorcery opened the pull request:
When Host header is forwarded as-is, service meshes such as Istio will erroneously send the request to the the host specified there instead of the intended target.

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

jjoyce0510 reviewed (approved):
This makes sense to me. Thank you @codesorcery !

codesorcery commented:
Hey @jjoyce0510, thanks for reviewing the PR! I fixed some mistakes, rebased the commit and locally made sure that the build and tests runs correctly (haven't used Java and Gradle in a while).
